### ISSUE TITLE
Possibly incorrect Union declarations

### ISSUE THREAD

**Issue by Dynosaur** [pelotom/runtypes#207]
**[Example repo](https://github.com/Dynosaur/runtypes-arbitrary-union-example/tree/master)**

After some quick testing, it appears that the following code works as intended, in versions 5.2.0 and 5.0.2.
```
import { Union, Literal } from 'runtypes';

const stringsToBecomeLiterals: string[] = ['foo', 'bar'];
const literals = stringsToBecomeLiterals.map(s => Literal(s));

const union = Union(...literals) // "foo" | "bar"
```
However, in both versions, I get this error:
```
Expected 1-20 arguments, but got 0 or more.ts(2556)
union.d.ts(105, 45): An argument for 'A' was not provided.
```
However, if typescript errors are ignored during compilation and is allowed to run, it runs as intended with no runtime errors. This leads me to believe that the `lib/types/union.d.ts` is incorrectly typed.

Not sure though if maybe I just got lucky and it works somehow, but isn't supposed to. Is that a bug or a feature? 😉

**Comment by yuhr** [comment]
Yes, this is a known limitation of current implementation of `Union`, and it also applies to `Intersect`, `Tuple`, etc. This is resolved by #166 which has been merged already, but this will introduce a breaking change in the static typing of runtype interfaces. So please wait the next major version 6!

**Comment by yuhr** [comment]
Let's track this at #38.
